compiler/ccgtypes.nim

```diff
@@ -610,12 +610,13 @@ proc getRecordDesc(m: BModule, typ: PType, name: Rope,
                       [getTypeDescAux(m, typ[0].skipTypes(skipPtrs), check)])
       if typ.isException and m.config.exc == excCpp:
         appcg(m, result, "virtual void raise() { throw *this; }$n", []) # required for polymorphic exceptions
-        if typ.sym.magic == mException:
-          # Add cleanup destructor to Exception base class
-          appcg(m, result, "~$1();$n", [name])
-          # define it out of the class body and into the procs section so we don't have to
-          # artificially forward-declare popCurrentExceptionEx (very VERY troublesome for HCR)
-          appcg(m, cfsProcs, "inline $1::~$1() {if(this->raiseId) #popCurrentExceptionEx(this->raiseId);}$n", [name])
+        when false:
+          if typ.sym.magic == mException:
```

Comment:
I have tried this PR and works really well with one change. 
Line:
```
appcg(m, result, "virtual void raise() { throw *this; }$n", []) # required for polymorphic exceptions
```
needs to be moved into `when false:`